File: solutions_5652388522229760_1/C++/m11121/count.cpp
#include <iostream>
#include <cstdio>
using namespace std;

int main()
{
	int t;
	cin>>t;
	int ic=1;
	while(t--)
	{
		long long n;
		cin>>n;
		cout<<"Case #"<<ic++<<": ";
		char mp[10]{};
		int i,num=10;
		long long u,t;
		for( i=1;num&&i<1000000;++i)
		{
			t= u = i*n;
			while(u)
			{
				if(mp[u%10]==0)
				{
					mp[u%10]=1;
					--num;
				}
				u/=10;
			}

		}
		if(num==0) cout<<t<<endl;
		else cout<<"INSOMNIA"<<endl;
	}
	return 0;
}

File: solutions_5652388522229760_1/Python/Ranmocy/count.py
def solve(num):
    seen = [False for i in range(10)]
    seen_count = 0
    current = num
    for i in xrange(1,1000): # if not found, may be never found
        for c in str(current):
            digit = int(c)
            if not seen[digit]:
                seen[digit] = True
                seen_count += 1
            if seen_count == 10:
                return current
        current += num

    return 'INSOMNIA'

with open('count.in', 'r') as fin:
    with open('count.out', 'w') as fout:
        T = int(fin.readline())
        for i in xrange(1, T+1):
            N = int(fin.readline())
            fout.write('Case #{0}: {1}\n'.format(i, solve(N)))
